From 1d48451145d8351349ecdfa6a13ea422a4480cd3 Mon Sep 17 00:00:00 2001
From: Tony Cook <tony@develop-help.com>
Date: Wed, 9 Apr 2014 11:58:33 +1000
Subject: [PATCH] [perl #121505] add -fwrapv to ccflags for gcc 4.9 and later

But skip adding it if -fwrapv is already provided, or if -fno-wrapv or
-fsanitize=undefined is supplied in ccflags.
---
 Configure |   16 ++++++++++++++++
  1 file changed, 16 insertions(+)
  
diff --git a/Configure b/Configure
index 15b3da1..56ce12d 100755
--- a/Configure	2014-05-17 21:40:09.000000000 +0000
+++ b/Configure	2014-05-17 21:42:36.795497307 +0000
@@ -4518,6 +4518,22 @@
     $rm -f try try.*
 esac
 
+# gcc 4.9 by default does some optimizations that break perl.
+# see ticket 121505.
+#
+# The -fwrapv disables those optimizations (and probably others,) so
+# for gcc 4.9 (and later, since the optimizations probably won't go
+# away), add -fwrapv unless the user requests -fno-wrapv, which
+# disables -fwrapv, or if the user requests -fsanitize=undefined,
+# which turns the overflows -fwrapv ignores into runtime errors.
+case "$gccversion" in
+4.9.*|4.1[0-9].*|[5-9].*)
+    case "$ccflags" in
+    *-fno-wrapv*|*-fsanitize=undefined*|*-fwrapv*) ;;
+    *) ccflags="$ccflags -fwrapv" ;;
+    esac
+esac
+
 : What should the include directory be ?
 echo " "
 $echo $n "Hmm...  $c"
--
1.7.10.4
